### DISCUSSION TITLE
Is there any way to add custom actions?

### DISCUSSION THREAD

**Discussion by brunoprietog** [excid3/madmin#135] [Q&A] [answered]
Hi,
I would like to know if there is a possibility to add custom actions, and if possible, how to do it?
Thanks!

**Comment by cjavdev** [comment]
I used this PR for guidance and was able to get custom actions working: https://github.com/excid3/madmin/commit/2351ae087bf4cbe871dc80b48efade224f37fe6b

**Comment by cjavdev** [comment, marked as answer]
Put a call to `member_action` in your Resource that will render some HTML into the detail view of the resource.

```rb
  member_action do
    button_to "Fetch", main_app.fetch_madmin_video_path(@record), method: :patch, data: { turbo_confirm: "Are you sure?" }, class: "btn btn-primary"
  end
```

Put a route in your config/routes/madmin.rb routes:

```rb
  resources :videos do
    member do
      patch :fetch
    end
  end
```

Add the controller method to the madmin controller

```rb
    def fetch
      @record.fetch!
      redirect_to main_app.madmin_video_path(@record), notice: "Video fetched!"
    end
```

**Reply by excid3** [reply]
I need to add this to the docs. 👍

**Reply by cjavdev** [reply]
Happy to help add docs. I wasn't sure if that was the official direction you wanted to go and I'm using edge now :). Also, wasn't sure if you had plans for list actions (django admin has some pretty interesting tools for executing an action for many records that you select with a checkbox in the admin). Member actions are all I need for now and I'm sure it solves most use-cases.

**Reply by excid3** [reply]
Yeah! I definitely want to do that. I think they might get a little more complicated since they'll need to have checkboxes for specific records and all records, but I just hadn't gotten to that yet.

**Reply by Swalden** [reply]
I've noticed the latest gem doesn't doesn't include these commits yet: https://rubygems.org/gems/madmin/versions

Is the best way to then include this way: in gemfile: gem 'madmin', git: "https://github.com/excid3/madmin"

**Reply by excid3** [reply]
Doh, I'll have to cut a new release soon. For now, using it from git in your Gemfile will do the trick. 👍 

You can also use the github shortcut `gem "madmin", github: "excid3/madmin"`
